src/components/KanbanComponent/KanbanComponent.tsx

```diff
@@ -26,29 +13,125 @@ import { CardComponent } from '../CardComponent';
+  updatedAt: string;
+};
+
+export type ReviewType = {
+  author: {
+    login: string;
+  };
+  body: string;
+  comments: {
+    nodes: CommentType[];
+  };
+  createdAt: string;
+  state: string;
+  updatedAt: string;
+};
+
+export type LabelType = {
+  color: string;
+  name: string;
+};
+
+export type PRType = {
+  author: {
+    login: string;
+  };
+  body: string;
+  createdAt: string;
+  number: number;
+  labels: {
+    nodes: LabelType[];
+  };
+  reviews: {
+    nodes: ReviewType[];
+  };
+  state: string;
+  title: string;
+  updatedAt: string;
+  url: string;
+};
+
+export type RepoType = {
+  repository: string;
+  data: PRType[];
+};
+
 export const KanbanComponent = () => {
-  const [query, setQuery] = useState<QueryType>();
+  const [query, setQuery] = useState<PRType>();
+  const [userRepos, setUserRepos] = useState<RepoType[]>();
+  const [userRepoNames, setUserRepoNames] = useState<String[]>();
   const [sideDrawOpen, setSideDrawOpen] = useState(false);
+  const [userRepoView, setUserRepoView] = useState(0);
+  const [username, setUsername] = useState<string | undefined>('');
+
+  const KanbanColumnBody = (pullRequests: PRType[]) => {
+    console.log('pullrequest', pullRequests);
 
-  const KanbanColumnBody = (pullRequests: any[]) => {
     return (
       <Container style={{ maxHeight: 700, overflow: 'auto' }}>
-        {pullRequests.map((item: any) => (
-          <CardComponent
-            data={item}
-            key={item.id}
-            onQuery={setQuery}
-            onSideDrawOpen={setSideDrawOpen}
-          />
-        ))}
+        {pullRequests &&
+          pullRequests.map((item: PRType, id: number) => (
+            <CardComponent
+              data={item}
+              key={id}
+              onQuery={setQuery}
+              onSideDrawOpen={setSideDrawOpen}
+            />
+          ))}
       </Container>
     );
   };
 
+  const requestOptions = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: '',
+  };
+
+  const ghecAuthApi = useApi(githubAuthApiRef);
+  useEffect(() => {
+    async function fetchData() {
+      const backstageUserIdentity = await ghecAuthApi.getProfile();
+      setUsername(backstageUserIdentity?.displayName);
+      requestOptions.body = JSON.stringify({ user_id: backstageUserIdentity?.displayName });
+
+      console.log('reqoptions', requestOptions);
+      await fetch('http://localhost:7007/api/pr-tracker-backend/get-user-repos', requestOptions)
+        .then((response) => response.json())
+        .then((data) => {
+          console.log('DATA', data);
+          setUserRepos(data);
+
+          const repoNames: String[] = [];
+          data.map((repo: RepoType) => {
+            repoNames.push(repo.repository);
+          });
+
+          console.log('REPONAMES', repoNames);
+
+          setUserRepoNames(repoNames);
+        });
+
+      console.log(3);
+    }
+    fetchData();
+    console.log('USER REPOS', userRepos);
+  }, []);
```

Comment:
I'd suggest using `useAsync` instead
```suggestion
  const {value: {repoNames, userRepos}, loading, error} = useAsync(async () => {

        const backstageUserIdentity = await ghecAuthApi.getProfile();
        setUsername(backstageUserIdentity?.displayName);
        requestOptions.body = JSON.stringify({ user_id: backstageUserIdentity?.displayName });

        console.log('reqoptions', requestOptions);
        const repoNames = await fetch('http://localhost:7007/api/pr-tracker-backend/get-user-repos', requestOptions)
                .then((response) => response.json())
                .then((data) => {
                    console.log('DATA', data);

                    const repoNames: String[] = [];
                    data.map((repo: RepoType) => {
                        repoNames.push(repo.repository);
                    });

                    console.log('REPONAMES', repoNames);

                    return {repoNames, userRepos};
                });

        console.log(3);
        console.log('USER REPOS', userRepos);
    })

```

Reply:
You don't HAVE to make this change, but my team heavily prefers to use `useAsync` over `useEffect`. The suggestion may not work out of the box since I haven't tested it yet